docs/src/passing-in-data.rst

```diff
@@ -0,0 +1,219 @@
+=======================================
+
+You can specify a context variable either as
+part of a |toolkit| command,
+or in a **context** section of *cdk.json*.
+
+To pass the value of a command-line context value to your app,
+run a |toolkit| command using the **--context** (**-c**) option,
+as shown in the following example.
+
+.. code-block:: sh
+
+    cdk synth -c bucket_name=mygroovybucket
+
+To specify the same context variable in *cdk.json*:
+
+.. code-block:: json
+
+    {
+        "context": {
+            "bucket_name": "myotherbucket"
+        }
+    }
+
+To get the value of a context variable in your app,
+use code like the following,
+which gets the value of the context variable **bucket_name**.
+
+.. code-block:: ts
+
+    const bucket_name string = this.getContext("bucket_name");
+
+.. _passing_in_data_from_env_vars:
+
+Getting a Value from an Environment Variable
+============================================
+
+To get the value of an environment variable,
+use code like the following,
+which gets the value of the environment variable **MYBUCKET**.
+
+.. code-block:: ts
+
+    const bucket_name = process.env.MYBUCKET;
+
+.. _passing_in_data_from_ssm:
+
+Getting a Value from an SSM Store Variable
+==========================================
+
+To get the value of an SSM parameter store variable,
+use code like the following,
+which uses the value of the SSM variable **my-awesome-value**.
+
+.. code-block:: ts
+
+    const myvalue = new SSMParameterProvider(this).getString("my-awesome-value");
+
+See the :doc:`context` topic for information about the :py:class:`SSMParameterProvider <@aws-cdk/cdk.SSMParameterProvider>`.
+
+.. _passing_in_data_between_stacks:
+
+Passing in Data From Another Stack
+==================================
+
+You can pass data from one stack to another stack in the same app
+by using the **export** method in one stack and the **import** method in the other stack.
+
+The following example creates a bucket on one stack
+and passes a reference to the other stack through an interface.
+
+First create a stack with a bucket.
+The stack includes a a property we use to pass the bucket's properties to the other stack.
+Note how we use the **export** method on the bucket to get it's properties and save them
+in the stack property.
+
+.. code-block:: ts
+
+    class HelloCdkStack extends cdk.Stack {
+        // Property that defines the stack you are exporting from
+        public readonly myBucketRefProps: s3.BucketRefProps;
+
+	constructor(parent: cdk.App, name: string, props?: cdk.StackProps) {
+            super(parent, name, props);
+
+            const mybucket = new s3.Bucket(this, "MyFirstBucket");
+
+            // Save bucket's *BucketRefProps*
+            this.myBucketRefProps = mybucket.export();
+	}
+    }
+
+Next create another stack that gets a reference to the other bucket
+from the properties passed in through the constructor.
+
+.. code-block:: ts
+
+    // The class for the other stack
```

Comment:
I'd like it better if you put the `XferBucketProps` along with this class here, and call it `MyCdkStackProps`. Two reasons:

* Naming: this is the convention.
* Order: You're referring to that interface here, but only defining it in a completely different block. That doesn't help scannability.